report/topology.go

```diff
@@ -20,16 +20,15 @@ func MakeTopology() Topology {
 	}
 }
 
-// WithNode produces a topology from t, with nmd added under key nodeID; if a
-// node already exists for this key, nmd is merged with that node. Note that a
-// fresh topology is returned.
-func (t Topology) WithNode(nodeID string, nmd Node) Topology {
+// AddNode adds node to the topology under key nodeID; if a
+// node already exists for this key, nmd is merged with that node.
+// The same topology is returned to enable chaining.
```

Comment:
Can you add some language here that indicates this method is different from all the other similar methods (it mutates the Topology) to solve issues of GC pressure? Otherwise it's liable to be removed in a refactor.